Sources/YUL/AST.swift

```diff
@@ -0,0 +1,342 @@
+//
+//  AST.swift
+//  YUL
+//
+//  Created by Yicheng Luo on 11/16/18.
+//
+
+import Foundation
+
+public typealias Identifier = String
+
+public enum Literal: CustomStringConvertible {
+  case num(Int)
+  case string(String)
+  case bool(Bool)
+  case decimal(Int, Int)
+  case hex(String)
+
+  public var description: String {
+    switch self {
+    case .num(let i):
+      return String(i)
+    case .string(let s):
+      return "\"\(s)\""
+    case .bool(let b):
+      return b ? "1" : "0"
+    case .decimal(let (n1, n2)):
+      return "\(n1).\(n2)"
+    case .hex(let h):
+      return h
+    }
+  }
+}
+
+public struct Block: CustomStringConvertible {
+  public var statements: [Statement]
+
+  public init(_ statements: [Statement]) {
+    self.statements = statements
+  }
+
+  public var description: String {
+    let statement_description = self.statements.map({ s in
```

Comment:
1. `self` is unnecessary here, it's implicit anyway.
2. Don't bind the parameter for a 1-line function body. Just do 

```swift
let descriptions = statements.map { $0.description }.joined(separator: "\n")
```

Reply:
(I also prefer the brace-less syntax.